internal/handlers/forwarder.go

```diff
@@ -50,6 +51,28 @@ func NewForwarder(f *Forwarder) *Forwarder {
 	return f
 }
 
+type bufPool struct {
+	pool sync.Pool
+}
+
+func (b *bufPool) Put(buf []byte) {
+	b.pool.Put(&buf)
+}
+
+func (b *bufPool) Get() []byte {
+	bufp := b.pool.Get().(*[]byte)
+	return *bufp
+}
+
+func newBufPool(sz int) httputil.BufferPool {
+	return &bufPool{pool: sync.Pool{
+		New: func() interface{} {
+			buf := make([]byte, sz)
+			return buf
+		},
+	}}
+}
+
```

Comment:
Just in case someone else need the helper:

```suggestion
type bufPool struct {
	sz   int
	pool sync.Pool
}

func (b *bufPool) Put(buf []byte) {
	if cap(buf) < b.sz || cap(buf) > b.sz*2 {
		// Buffer too small or will likely leak memory after being expanded.
		// Drop it.
		return
	}
	b.pool.Put(&buf)
}

func (b *bufPool) Get() []byte {
	bufp := b.pool.Get().(*[]byte)
	return (*bufp)[:b.sz]
}

func newBufPool(sz int) httputil.BufferPool {
	return &bufPool{sz: sz, pool: sync.Pool{
		New: func() interface{} {
			buf := make([]byte, sz)
			return buf
		},
	}}
}
```